packages/ui-components/src/components/Badge/Badge.component.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Juno contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from "react"
+
+import { Icon, KnownIcons, KnownIconsEnum } from "../Icon/Icon.component"
+import { BadgeProps, VariantType } from "./Badge.types"
+
+// BADGE STYLES
+
+const badgeBaseStyles = `
+  jn-rounded
+  jn-text-sm
+  jn-text-theme-default
+  jn-py-0.5
+  jn-px-1
+  jn-justify-center
+  jn-items-center
+`
+
+const badgeVariantStyles = {
+  default: "jn-bg-theme-badge-default",
+  info: "jn-bg-theme-info/25",
+  success: "jn-bg-theme-success/25",
+  warning: "jn-bg-theme-warning/25",
+  danger: "jn-bg-theme-danger/25",
+  critical: "jn-bg-theme-danger/70 jn-text-theme-high",
+  error: "jn-bg-theme-error/25",
+}
+
+const getVariantStyle = (variant: VariantType): string => badgeVariantStyles[variant] || badgeVariantStyles.default
+
+// ICON UTILS
+
+const iconStyles = "jn-mr-1 jn-items-center"
+
+// Get icon based on the provided variant or icon string
+const getIcon = (icon: boolean | KnownIcons | undefined, variant: VariantType): KnownIcons | undefined => {
```

Comment:
lets discuss this one on personal call, I have few suggestions :)

Reply:
Resolution described [here](https://github.com/cloudoperators/juno/pull/514#discussion_r1800619164).